src/nifake/unit_tests/test_session.py

```diff
@@ -157,6 +157,28 @@ def test_simple_function(self):
             session.simple_function()
             self.patched_library.niFake_PoorlyNamedSimpleFunction.assert_called_once_with(matchers.ViSessionMatcher(SESSION_NUM_FOR_TEST))
 
+    def test_self_test(self):
+        self.patched_library.niFake_self_test.side_effect = self.side_effects_helper.niFake_self_test
+        test_error_code = 0
+        self.side_effects_helper['self_test']['selfTestResult'] = test_error_code
+        self.side_effects_helper['self_test']['selfTestMessage'] = ''
+        with nifake.Session('dev1') as session:
+            session.self_test()
+
+    def test_self_test_fail(self):
+        self.patched_library.niFake_self_test.side_effect = self.side_effects_helper.niFake_self_test
+        test_error_code = 1
+        test_error_message = 'error message'
+        self.side_effects_helper['self_test']['selfTestResult'] = test_error_code
+        self.side_effects_helper['self_test']['selfTestMessage'] = test_error_message
+        with nifake.Session('dev1') as session:
+            try:
+                session.self_test()
+                assert False
+            except Exception as e:
```

Comment:
I am not sure what to do here. We do not export the individual exceptions from errors, and the other exceptions inherit from `Exception`, not the driver `Error`. I see two ways to fix this:
* Export all the different exceptions
* All driver specific exceptions inhering from the base driver exception called `Error`

What do others think?

Reply:
``` python
import nifake.errors

try:
    blah()
except nifake.errors.SelfTestFailureError as e:
    print(e)
```
works.

Is this acceptable for customers?